### PR TITLE
worker: Rename worker_interface to worker

### PR DESCRIPTION
**Pull request by gaocegege**
worker_interface is a little misunderstanding, since it is not just a package for interface.

/assign @YujiOshima 

Signed-off-by: Ce Gao <gaoce@caicloud.io>

### CODE REVIEW

**Comment by YujiOshima** [comment]
@gaocegege The worker is an instance of evaluating a trial.
So the worker is a k8s job, a TF Job etc.
How about `worker-controller`?

**Comment by gaocegege** [comment]
I think worker suits it, because it represents the worker to run these trials :thinking: 

WDYT @ddysher

**Comment by ddysher** [comment]
conceptual-wise, i agree with @YujiOshima that `worker` is more like a controller; however, using worker_controller is quite verbose IMO; so my suggestion is to simply use `worker` as proposed in this PR.

File: cmd/manager/main.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ type studyCh struct {
 	addMetricsCh chan string
 }
 type server struct {
-	wIF         worker_interface.WorkerInterface
+	wIF         worker.WorkerInterface
```
Comment (ddysher) [inline cmd/manager/main.go]: Can we change `WorkerInterface` to `Interface`, per
https://github.com/golang/go/wiki/CodeReviewComments#package-names

example:
https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/cloud.go

**Comment by ddysher** [comment]
LGTM

@YujiOshima do you want to take another look?

**Comment by YujiOshima** [comment]
@ddysher Thanks! LGTM the name worker.Interface !
/approve

**Comment by gaocegege** [comment]
I am not sure why tide does not merge it automatically. Same as https://github.com/kubeflow/tf-operator/pull/557

**Comment by gaocegege** [comment]
It is too hack.. We can not merge a PR without lgtm although we have already added approve.

**Pull request merged**